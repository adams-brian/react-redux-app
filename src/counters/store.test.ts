import { actionCreators, ADD_COUNTER, COUNTERS_ERROR, COUNTERS_LOADED,
  COUNTERS_UPDATED, DECREMENT, errorReducer, INCREMENT,
  listReducer, LOAD_COUNTERS, loadedReducer, REMOVE_COUNTER, RESET } from "./store";

const counters = [2,-2];

describe('store', () => {
  
  describe('actions', () => {

    it('should export action types', () => {
      expect(ADD_COUNTER).toMatchSnapshot();
      expect(COUNTERS_ERROR).toMatchSnapshot();
      expect(COUNTERS_LOADED).toMatchSnapshot();
      expect(COUNTERS_UPDATED).toMatchSnapshot();
      expect(DECREMENT).toMatchSnapshot();
      expect(INCREMENT).toMatchSnapshot();
      expect(LOAD_COUNTERS).toMatchSnapshot();
      expect(REMOVE_COUNTER).toMatchSnapshot();
      expect(RESET).toMatchSnapshot();
    });

  });

  describe('actionCreators', () => {
    
    it('should export actionCreators', () => {
      expect(actionCreators.addCounter()).toMatchSnapshot();
      expect(actionCreators.countersError(new Error('error message'))).toMatchSnapshot();
      expect(actionCreators.countersLoaded()).toMatchSnapshot();
      expect(actionCreators.countersUpdated(counters)).toMatchSnapshot();
      expect(actionCreators.decrement(1)).toMatchSnapshot();
      expect(actionCreators.increment(2)).toMatchSnapshot();
      expect(actionCreators.loadCounters()).toMatchSnapshot();
      expect(actionCreators.removeCounter(3)).toMatchSnapshot();
      expect(actionCreators.reset(4)).toMatchSnapshot();
    });

  });

  describe('listReducer', () => {

    it('should increment', () => {
      expect(listReducer(counters, actionCreators.increment(0))).toEqual([3,-2]);
      expect(listReducer(counters, actionCreators.increment(1))).toEqual([2,-1]);
    });

    it('should decrement', () => {
      expect(listReducer(counters, actionCreators.decrement(0))).toEqual([1,-2]);
      expect(listReducer(counters, actionCreators.decrement(1))).toEqual([2,-3]);
    });

    it('should reset', () => {
      expect(listReducer(counters, actionCreators.reset(0))).toEqual([0,-2]);
      expect(listReducer(counters, actionCreators.reset(1))).toEqual([2,0]);
    });

    it('should add counter', () => {
      expect(listReducer(undefined, actionCreators.addCounter())).toEqual([0]);
      expect(listReducer(counters, actionCreators.addCounter())).toEqual([2,-2,0]);
    });

    it('should remove counter', () => {
      expect(listReducer(counters, actionCreators.removeCounter(0))).toEqual([-2]);
      expect(listReducer(counters, actionCreators.removeCounter(1))).toEqual([2]);
    });

    it('should update counters', () => {
      const update = [5,1,4,2,3];
      expect(listReducer(undefined, actionCreators.countersUpdated(update))).toEqual(update);
      expect(listReducer(counters, actionCreators.countersUpdated(update))).toEqual(update);
    });

  });

  describe('loadedReducer', () => {

    it('should default to false', () => {
      expect(loadedReducer(undefined, actionCreators.addCounter())).toBe(false);
    });

    it('should return the state when not CountersLoaded', () => {
      expect(loadedReducer(false, actionCreators.addCounter())).toBe(false);
      expect(loadedReducer(true, actionCreators.addCounter())).toBe(true);
    });

    it('should return true when CountersLoaded', () => {
      expect(loadedReducer(undefined, actionCreators.countersLoaded())).toBe(true);
      expect(loadedReducer(false, actionCreators.countersLoaded())).toBe(true);
      expect(loadedReducer(true, actionCreators.countersLoaded())).toBe(true);
    });

  });

  describe('errorReducer', () => {

    it('should default to empty string', () => {
      expect(errorReducer(undefined, actionCreators.addCounter())).toEqual('');
    });

    it('should return the state when not CountersError', () => {
      const message = 'the error message';
      expect(errorReducer('', actionCreators.addCounter())).toEqual('');
      expect(errorReducer(message, actionCreators.addCounter())).toEqual(message);
    });

    it('should return the message from the Error when CountersError', () => {
      const err = new Error('the error message');
      expect(errorReducer(undefined, actionCreators.countersError(err))).toEqual(err.message);
      expect(errorReducer('different message', actionCreators.countersError(err))).toEqual(err.message);
    });

  });

});
