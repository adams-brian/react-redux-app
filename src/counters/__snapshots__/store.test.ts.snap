// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store actionCreators should export actionCreators 1`] = `
Object {
  "type": "[Counters] ADD COUNTER",
}
`;

exports[`store actionCreators should export actionCreators 2`] = `
Object {
  "payload": [Error: error message],
  "type": "[Counters] COUNTERS ERROR",
}
`;

exports[`store actionCreators should export actionCreators 3`] = `
Object {
  "type": "[Counters] COUNTERS LOADED",
}
`;

exports[`store actionCreators should export actionCreators 4`] = `
Object {
  "payload": Array [
    2,
    -2,
  ],
  "type": "[Counters] COUNTERS UPDATED",
}
`;

exports[`store actionCreators should export actionCreators 5`] = `
Object {
  "payload": 1,
  "type": "[Counters] DECREMENT",
}
`;

exports[`store actionCreators should export actionCreators 6`] = `
Object {
  "payload": 2,
  "type": "[Counters] INCREMENT",
}
`;

exports[`store actionCreators should export actionCreators 7`] = `
Object {
  "type": "[Counters] LOAD COUNTERS",
}
`;

exports[`store actionCreators should export actionCreators 8`] = `
Object {
  "payload": 3,
  "type": "[Counters] REMOVE COUNTER",
}
`;

exports[`store actionCreators should export actionCreators 9`] = `
Object {
  "payload": 4,
  "type": "[Counters] RESET",
}
`;

exports[`store actions should export action types 1`] = `"[Counters] ADD COUNTER"`;

exports[`store actions should export action types 2`] = `"[Counters] COUNTERS ERROR"`;

exports[`store actions should export action types 3`] = `"[Counters] COUNTERS LOADED"`;

exports[`store actions should export action types 4`] = `"[Counters] COUNTERS UPDATED"`;

exports[`store actions should export action types 5`] = `"[Counters] DECREMENT"`;

exports[`store actions should export action types 6`] = `"[Counters] INCREMENT"`;

exports[`store actions should export action types 7`] = `"[Counters] LOAD COUNTERS"`;

exports[`store actions should export action types 8`] = `"[Counters] REMOVE COUNTER"`;

exports[`store actions should export action types 9`] = `"[Counters] RESET"`;
